thin-replica-server/proto/state_snapshot_concord.proto

```diff
@@ -0,0 +1,54 @@
+// Copyright 2021 VMware, all rights reserved
+//
+// Concord Client's StateSnapshot Service
+
+syntax = "proto3";
+package com.vmware.concord.state_snapshot_concord;
+
+option java_package = "com.vmware.concord.state_snapshot_concord";
+
+// The Concord StateSnapshot service can be used by the concord client to 
+// retrieve state snapshot from a concord cluster
+// See `client/proto/state_snapshot.proto` for definition of state snapshot.
+
+service StateSnapshot {
+  // Stream a specific state snapshot in a resumable fashion as a finite stream of key-values.
+  // Key-values are streamed with lexicographic order on keys.
+  // Errors:
+  // NOT_FOUND: if a state snapshot with the requested ID is not (or no longer) available.
+  //            Applications are advised to retry initialization by fetching a new snapshot ID.
```

Comment:
We could change `application` here, because it is the client that connects to the replica.

Reply:
Thanks for catching, updated.